pkg/stringutils/stringutils.go

```diff
@@ -33,14 +36,49 @@ func GenerateRandomASCIIString(n int) string {
 // Ellipsis truncates a string to fit within maxlen, and appends ellipsis (...).
 // For maxlen of 3 and lower, no ellipsis is appended.
 func Ellipsis(s string, maxlen int) string {
-	r := []rune(s)
-	if len(r) <= maxlen {
+	byteLen := len(s)
+	if byteLen == utf8.RuneCountInString(s) {
+		r := []rune(s)
+		if byteLen <= maxlen {
+			return s
+		}
+		if maxlen <= 3 {
+			return string(r[:maxlen])
+		}
+		return string(r[:maxlen-3]) + "..."
+	}
+
+	var display []int
+	displayLen := 0
+	rs := []rune(s)
+	for _, r := range rs {
+		//In a broad sense, wide characters include East Asian Wide, East Asian Fullwidth, and East Asian Ambiguous,
+		//see http://unicode.org/reports/tr11/
+		kind := width.LookupRune(r).Kind()
+		if kind == width.EastAsianAmbiguous || kind == width.EastAsianWide || kind == width.EastAsianFullwidth {
```

Comment:
Perhaps this should be used to a utility function, something like:

```go
// charWidth returns the number of horizontal positions a character occupies,
// and is used to account for wide characters when displaying strings.
//
// In a broad sense, wide characters include East Asian Wide, East Asian Full-width,
// and East Asian Ambiguous (see http://unicode.org/reports/tr11/).
func charWidth(r rune) int {
	switch width.LookupRune(r).Kind() {
	case width.EastAsianAmbiguous, width.EastAsianWide, width.EastAsianFullwidth:
		return 2
	default:
		return 1
	}
}
```

Then this loop could be something like;

```go
	var (
		display      []int
		displayWidth int
	)
	rs := []rune(s)
	for _, r := range rs {
		cw := charWidth(r)
		displayWidth += cw
		display = append(display, cw)
	}
```

Reply:
OK , I will improve the code after we move the function to CLI repository.

Reply:
Thanks! Are you interested in working on that move?

Reply:
It will be a great pleasure ：)